src/CodeGen/ODataT4CodeGenerator.ttinclude

```diff
@@ -3491,7 +3514,34 @@ namespace <#= fullNamespace #>
                 global::System.Xml.XmlReader reader = CreateXmlReader(Edmx);
                 try
                 {
-                    return global::Microsoft.OData.Edm.Csdl.CsdlReader.Parse(reader);
+                    global::Microsoft.OData.Edm.Csdl.CsdlReaderSettings edmxReaderSettings = new global::Microsoft.OData.Edm.Csdl.CsdlReaderSettings()
+                    {
+<#+
+            if (this.context.IgnoreUnexpectedElementsAndAttributes)
+            {
+#>
+                        IgnoreUnexpectedAttributesAndElements = true
+<#+
+            }
+            else
+            {
+#>
+                        IgnoreUnexpectedAttributesAndElements = false
+<#+
+            }
+#>
+                    };
+                    
+                    global::System.Collections.Generic.IEnumerable<global::Microsoft.OData.Edm.Validation.EdmError> errors;
+                    global::System.Collections.Generic.IEnumerable<global::Microsoft.OData.Edm.IEdmModel> references = global::System.Linq.Enumerable.Empty<global::Microsoft.OData.Edm.IEdmModel>();
+                    global::Microsoft.OData.Edm.IEdmModel edmModel;
+                    
+                    if (!global::Microsoft.OData.Edm.Csdl.CsdlReader.TryParse(reader, references, edmxReaderSettings, out edmModel, out errors))
+                    {

```

Comment:
here, too, should be able to use:
       public static bool TryParse(XmlReader reader, bool ignoreUnexpectedAttributesAndElements, out IEdmModel model, out IEnumerable<EdmError> errors)

Reply:
:+1: